src/Tasks/FileIO/GetFileHash.cs

```diff
@@ -115,6 +118,13 @@ public override bool Execute()
             return !Log.HasLoggedErrors;
         }
 
+        private readonly CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
```

Comment:
nit: `CancellationTokenSource` implements `IDisposable` and should have its `Dispose` called when we're done with it. Non-blocking comment as we have this pattern in many other tasks and it's not really causing issues. There's a chance we will want to address this as we harden the code against leaks, though.

Reply:
I leave that to you.